force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -124,18 +128,15 @@ public with sharing class ClassesObjectsMethods {
     public static String findTheScientist() {
         // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
-
-        // The variable to store the concatenated string
-
-        // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
-        // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+        List<String> gNames = new List<String>();
+        for(String name : scientistNames){
+            if(name.containsIgnoreCase('g')== true){
+                gNames.add(name);
+            }
+        }
+        String aRoseByAnyOtherGname = String.join(gNames, ',');
+        aRoseByAnyOtherGname.removeEnd(',');
+        return aRoseByAnyOtherGname;
```

Comment:
Try to see if you can refactor this so that you are able to use the 'continue' keyword in this problem.